modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/ExecutorServiceExtension.java

```diff
@@ -0,0 +1,245 @@
+
+        IgniteUtils.closeAll(autoCloseableStream);
+    }
+
+    private static List<Field> collectFields(Class<?> testClass, boolean forStatic) {
+        return AnnotationSupport.findAnnotatedFields(
+                testClass,
+                InjectExecutorService.class,
+                field -> isStatic(field.getModifiers()) == forStatic,
+                HierarchyTraversalMode.TOP_DOWN
+        );
+    }
+
+    private static void checkFieldTypeIsSupported(Field field) {
+        if (!isFieldTypeIsSupported(field.getType())) {
+            throw new IllegalStateException(
+                    String.format("Unsupported field type: [field=%s, supportedFieldTypes=%s]", field, SUPPORTED_FIELD_TYPES)
+            );
+        }
+    }
+
+    private static boolean isFieldTypeIsSupported(Class<?> fieldType) {
+        for (Class<?> supportedFieldType : SUPPORTED_FIELD_TYPES) {
+            if (fieldType.equals(supportedFieldType)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static ExecutorService createExecutorService(Field field) {
+        InjectExecutorService injectExecutorService = field.getAnnotation(InjectExecutorService.class);
+
+        assert injectExecutorService != null : field;
+
+        return createExecutorService(injectExecutorService, field.getName(), field.getType(), field.getDeclaringClass());
+    }
+
+    private static ExecutorService createExecutorService(
+            InjectExecutorService injectExecutorService,
+            String fieldName,
+            Class<?> fieldType,
+            Class<?> testClass
+    ) {
+        int threadCount = injectExecutorService.threadCount();
+        String threadPrefix = threadPrefix(injectExecutorService, fieldName, fieldType);
+        ThreadOperation[] allowedOperations = injectExecutorService.allowedOperations();
+
+        ThreadFactory threadFactory = IgniteThreadFactory.withPrefix(threadPrefix, Loggers.forClass(testClass), allowedOperations);
+
+        if (fieldType.equals(ScheduledExecutorService.class)) {
+            return newScheduledThreadPool(threadCount == 0 ? 1 : threadCount, threadFactory);
+        } else if (fieldType.equals(ExecutorService.class)) {
+            return newFixedThreadPool(threadCount == 0 ? CPUS : threadCount, threadFactory);
+        }
+
+        throw new AssertionError(
+                String.format("Unsupported field type: [field=%s, supportedFieldTypes=%s]", fieldName, SUPPORTED_FIELD_TYPES)
+        );
+    }
+
+    private static String threadPrefix(InjectExecutorService injectExecutorService, String fieldName, Class<?> fieldType) {
+        String threadPrefix = injectExecutorService.threadPrefix();
+
+        if (threadPrefix != null && !"".equals(threadPrefix)) {
+            return threadPrefix;
+        }
+
+        return String.format("test-%s-%s", fieldType.getSimpleName(), fieldName);
```

Comment:
What if there are two fields named identically in a superclass and subclass? It seems their executors will get the same prefix

Reply:
You're right, I'll fix it a little.